alicloud/resource_alicloud_polardb_cluster.go

```diff
@@ -0,0 +1,417 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"strings"
+	"time"
+
+	"strconv"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/polardb"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudPolarDBCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudPolarDBClusterCreate,
+		Read:   resourceAlicloudPolarDBClusterRead,
+		Update: resourceAlicloudPolarDBClusterUpdate,
+		Delete: resourceAlicloudPolarDBClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"db_type": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"db_version": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"db_node_class": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"zone_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"cluster_charge_type": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{string(Postpaid), string(Prepaid)}, false),
+				Optional:     true,
+				Default:      Postpaid,
+			},
+			"renewal_status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  RenewNormal,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(RenewAutoRenewal),
+					string(RenewNormal),
+					string(RenewNotRenewal)}, false),
+			},
+			"auto_renew": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"auto_renew_period": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntInSlice([]int{1, 2, 3, 6, 12}),
+			},
+			"period": {
+				Type:         schema.TypeInt,
+				ValidateFunc: validation.IntInSlice([]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 12, 24, 36}),
+				Optional:     true,
+				Default:      1,
```

Comment:
I think the `renewal_status `, `auto_renew `, `auto_renew_period ` and `period ` should add diffsuppressfunc because of they are available when PrePaid.
You can refer to kvstore instance resource.

Reply:
done